sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -593,8 +593,14 @@ private[sql] object ParquetSchemaConverter {
        """.stripMargin.split("\n").mkString(" ").trim)
   }
 
-  def checkFieldNames(names: Seq[String]): Unit = {
-    names.foreach(checkFieldName)
+  def checkFieldNames(schema: StructType): Unit = {
+    schema.map(field => field.name -> field.dataType)
+      .foreach {
+        case (name, struct: StructType) =>
```

Comment:
I think you can traverse with `StructField`? e.g.)

```scala
  case field if field.dataType.isInstanceOf[StructType]  =>
    checkFieldName(name)
    checkFieldNames(struct)
  case field => checkFieldName(field.name)
```

Reply:
Nice suggestion, done